test/core.lua

```diff
@@ -344,6 +344,13 @@ local function test_macros()
         ["(-?>> :w (. {:w :x}) (. {:x :missing}) (. {:y :z}))"]=nil,
         ["(-?> [:a :b] (table.concat \" \"))"]="a b",
         ["(-?>> \" \" (table.concat [:a :b]))"]="a b",
+        -- let-open should close all handles and return result of body
+        ["(var (fh1 fh2) nil) [(let-open [f1 (io.tmpfile) f2 (io.tmpfile)]\
+          (set [fh1 fh2] [f1 f2]) (f1:write :asdf) (f1:seek :set 0) (f1:read :*a))\
+          (io.type fh1) (io.type fh2)]"]={"asdf", "closed file", "closed file"},
+        -- let-open should propogate errors and still close the file handles
+        ["(var fh nil) (local (ok msg) (pcall #(let-open [f (io.tmpfile)] (set fh f)\
+          (error :bork!)))) [(io.type fh) ok (msg:match :bork!)]"]={"closed file", false, "bork!"},
```

Comment:
We should add a test for multiple values here.

Reply:
That it returns multiple values? Or.. hmm, actually this raises a valid point. What `let-open` doesn't account for here is destructuring or multiple-value bindings like

```clojure
(let-open [(f1 f2) (values (io.open :a) (io.open :b))] xyz)
```

I can either add handling for that to make it continue to work, or I could make a no-destructuring assertion. Thoughts?

Reply:
Oh good point. I was thinking of multiple values in the body, but I hadn't considered multiple values in the binding. I think it's OK skip destructuring as long as we call it out as unsupported in the docs, and maybe add an explicit error message asserting that each binding is a `sym?`.

Reply:
So I initially agreed with this, but after letting it simmer for a week, I don't feel great about the `let-open` name if we're going to make the binding semantics stray from those of `let`. Clojure's `with-open` also only allows syms in the name position, no destructuring, so doing that here would track, but I think that flips my position back to pro `with-open` over `let-open` for the fennel equivalent. What do you think?

Reply:
Went ahead and added a `sym?` assertion and renamed it to `with-open`!